docs/knowledgebase/getting-started/index.md

```diff
@@ -74,119 +74,75 @@ rustup default stable
 
 Substrate uses [WebAssembly](https://webassembly.org/) (Wasm) to produce portable blockchain
 runtimes. You will need to configure your Rust compiler to use
-[`nightly` builds](https://doc.rust-lang.org/book/appendix-07-nightly-rust.html) to allow
-compiled substrate compatible runtimes to Wasm.
+[`nightly` builds](https://doc.rust-lang.org/book/appendix-07-nightly-rust.html) to allow you to
+compile Substrate runtime code to the Wasm target. There is
+[a GitHub Issue](https://github.com/paritytech/substrate/issues/1252) that describes this
+requirement in detail.
 
 #### Rust Nightly Toolchain
 
-Because the nightly toolchain is a moving target and receives daily changes the chance
-that some of them break the substrate build from time to time is non-negligible.
-
-Therefore it is advised to use a fixed nightly version rather than the latest one to
-build the runtime. You can install a specific version using this command:
+Developers building with Substrate should use a specific Rust nightly version that is known to be
+compatible with the version of Substrate they are using; this version will vary from project to
+project and different projects may use different mechanisms to communicate this version to
+developers. For instance, the Polkadot client specifies this information in its
+[release notes](https://github.com/paritytech/polkadot/releases). The Substrate Node Template uses
+an
+[init script](https://github.com/substrate-developer-hub/substrate-node-template/blob/master/scripts/init.sh)
+and
+[Makefile](https://github.com/substrate-developer-hub/substrate-node-template/blob/master/Makefile)
+to specify the Rust nightly version and encapsulate the following steps. Use Rustup to install the
+correct nightly:
 
 ```bash
 rustup install nightly-<yyyy-MM-dd>
```

Comment:
at the time of writing 2020-11-17, the current default nightly is 2020-10-05

Reply:
I don't want to add this because it creates the question: "the current default nightly that compiles _with what_?"

Reply:
I may be good idea to mention how to find the correct nightly for any case. You mention how to do it in Pokadot but why not mention that a way to find out is to take a look to the date of the release and use that date in the `nightly-<date of release>`

Reply:
From my POV that gets too deeply into supporting other projects. We need to be very careful about what we are documenting here. We are trying to teach people the best practices for working with Substrate code. I think that at this time it is very important to use a well-known nightly version that corresponds with whatever project you are using. I do not want to encourage people to release Substrate-based code without specifying an exact nightly to use, nor do I want to encourage our users to try to "guess" which nightly they should use. This is definitely an area where I am deciding _not_ to document every possible technical detail in order to encourage people to follow what I believe we should set up as best practice.

Reply:
I agree with almost everything except.
>encourage our users to try to "guess" which nightly they should use

I think that this a good to have as a way to trouble shoot. In the back it happens all the time, your machine doesn't has the same nightly as mine because we update at different time. My point is that it is a rule of thumb to get the date of the release and use it for nightly. I'm not talking about other projects I referring to every time we have a Substrate release.

Reply:
Wouldn't that be covered by this section? https://github.com/substrate-developer-hub/substrate-developer-hub.github.io/blob/df/fix/installation/docs/knowledgebase/getting-started/index.md#latest-nightly-for-substrate-master

Reply:
Nope, it could go there but it isn't covered. It would be suggesting them to use a very specific date which will work in 99.9999% of the cases. Substrate 2.0 was release in September 22 so using `nightly-2020-09-22` will work for them.

Reply:
No I'm sorry that is just not something I am comfortable supporting at this time.

Reply:
I don't get why? There is no reason for not doing it. It is correct to do it and will prevent us explaining the same thing every time or adding a lot of patches as you have been doing. Please give a good justification for not doing it.

Reply:
Just saying that you are not comfortable with something that doesn't requires support isn't a valid justification.

Reply:
I would say I provided that justification in my previous comment https://github.com/substrate-developer-hub/substrate-developer-hub.github.io/pull/772#discussion_r526205572 I don't feel comfortable setting a precedent that the Knowledge Base provides general support for _any_ or _all_ Substrate-based projects. I don't even really feel comfortable with the Knowledge Base having these instructions _at all_. I really believe this information belongs with the Node Template. I do not think it is good practice for a Substrate-based project to make a release without specifying the nightly it builds with, so therefore I do not feel comfortable providing any type of support or advice for these cases. It is "undefined behavior". That is my justification for not feeling comfortable supporting this.